### ISSUE TITLE
Notify devs that they should not merge a PR

### ISSUE THREAD

**Issue by jon-lewis** [near/pagoda-console#7]
Our current process allows anyone to merge a PR after they approve it. This may not work well in some cases (see below).

It would be nice to have a standardized way to properly communicate to the team if a PR is good to merge and, if not, why. It seems we have 2 use cases:
1. there is a complication with merging the PR and the PR should only be merged by the author
    * includes reasons like: many PRs need to be merged in coordination, some process outside of our automated processes must be done before or shortly after a merge, etc
    * I've been using the title prefix of `DO NOT MERGE: lorem ipsum` to indicate that PRs should not be merged without discussing with me because of some complication
2. the author of the PR would like to perform some cleanup or other tasks before the PR is merged and would like to merge themselves

It would be nice if Github allowed disabling the merge button in these cases but it doesn't seem like that is supported.

Another solution would be to use labels like `do_not_merge` and `author_merges` and generate a comment on the PR that gives more context on what each label means. We can also generate a comment on the PR with [this github action](https://docs.github.com/en/actions/managing-issues-and-pull-requests/commenting-on-an-issue-when-a-label-is-added) if the labels exist.

[Original conversation in Slack](https://pagodaplatform.slack.com/archives/C034SDNG8KT/p1659968192796499).
